urllib3/util/selectors.py

```diff
@@ -535,19 +535,44 @@ def close(self):
             super(KqueueSelector, self).close()
 
 
+if not hasattr(select, 'select'):  # Platform-specific: AppEngine
+    HAS_SELECT = False
+
+
+def _can_allocate(struct):
+    """ Checks that select structs can be allocated
+    by the underlying operating system, not just
+    advertised by the select module. We don't check
+    select() because we'll be hopeful that most 
+    platforms that don't have it available will not
+    advertise it. (ie: GAE) """
```

Comment:
Haha you can tell when I'm working on the GitHub web platform versus a real IDE. I'm sitting in my bed after just woken up making these changes and I have to run to work soon. :)